nodefs/simple_test.go

```diff
@@ -250,6 +250,38 @@ func TestReadDir(t *testing.T) {
 	posixtest.ReadDir(t, tc.mntDir)
 }
 
+func TestReadDirStress(t *testing.T) {
+	tc := newTestCase(t, true, true)
+	defer tc.Clean()
+	// (ab)use posixtest.ReadDir to create 110 test files
+	posixtest.ReadDir(t, tc.mntDir)
+
+	var wg sync.WaitGroup
+	stress := func(gr int) {
+		defer wg.Done()
+		for i := 1; i < 100; i++ {
+			f, err := os.Open(tc.mntDir)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			_, err = f.Readdirnames(-1)
+			if err != nil {
+				t.Errorf("goroutine %d iteration %d: %v", gr, i, err)
+				return
```

Comment:
nit, this leaks the fd, leading to error on unmount.

Reply:
Uh, right

Reply:
Fixed